eclipse-collections/src/main/java/org/eclipse/collections/impl/set/sorted/mutable/SortedSetAdapter.java

```diff
@@ -173,6 +173,16 @@ public boolean containsAll(Collection<?> collection)
     @Override
     public boolean equals(Object obj)
     {
+        if (obj == this)
+        {
+            return true;
+        }
+
+        if (!(obj instanceof Set))
+        {
+            return false;
+        }
+
         return this.delegate.equals(obj);
```

Comment:
Seems out of place

Reply:
@motlin what do you mean? Should I move `equals` and `hashCode` to AbstractCollectionAdapter maybe? But I wouldn't be able to test for instance of `Set` though